src/Symfony/Component/Workflow/Exception/UndefinedTransitionException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Exception;
+
+/**
+ * Thrown by Workflow when an undefined transition is applied on a subject.
+ */
+class UndefinedTransitionException extends SubjectTransitionException
```

Comment:
I created a dedicated exception for this case, so devs can do the following:
```php
try {
  $workflow->apply($subject, $transitionName);
} catch (UndefinedTransitionException $exception) {
  // throw hard
} catch (SubjectTransitionException $exception) {
  // return pretty response with a list of transition blockers.
}
```

Reply:
Ideally, I'd just throw a `LogicException` from all methods, that accept `$transitionName` as a parameter instead of introducing this Exception, but this would break the BC.

Reply:
On second thought, this `try..catch` is annoying, because it would force devs to always catch that exception just to rethrow it. Let me fix that.